cmd/zoekt-dynamic-indexserver/main.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2016 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This program manages a zoekt dynamic indexing deployment:
+// * listens to indexing commands
+// * reindexes specified repositories
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"time"
+)
+
+func loggedRun(cmd *exec.Cmd) (err error) {
+	outBuf := &bytes.Buffer{}
+	errBuf := &bytes.Buffer{}
+	cmd.Stdout = outBuf
+	cmd.Stderr = errBuf
+
+	log.Printf("run %v", cmd.Args)
+	if err = cmd.Run(); err != nil {
+		log.Printf("command %s failed: %v\nOUT: %s\nERR: %s",
+			cmd.Args, err, outBuf.String(), errBuf.String())
+		err = fmt.Errorf("command %s failed: %v", cmd.Args, err)
+	}
+
+	return
```

Comment:
using naked returns here is not normal go convention and is a bit of a footgun. You use that pattern in some other places as well. named returns are there to clarify godocs (and occasionally to simplify small functions). See https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

```suggestion
	if err := cmd.Run(); err != nil {
		log.Printf("command %s failed: %v\nOUT: %s\nERR: %s",
			cmd.Args, err, outBuf.String(), errBuf.String())
		return fmt.Errorf("command %s failed: %v", cmd.Args, err)
	}

	return nil
```